src/Exception/FrozenAttributeException.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1); // @codeCoverageIgnore
+
+namespace Rinq\Exception;
+
+use Rinq\Ident\Attribute;
+use RuntimeException;
+
+class FrozenAttributeException extends RuntimeException
+{
+    /**
+     * @param Attribute $attribute The attribute that could not be modified.
+     */
+    public function __construct(Attribute $attribute)
```

Comment:
The particular attribute that was frozen is not known when the failed update is for a remote session. It's simply not communicated back from the "server". This is why the [Go version](https://github.com/rinq/rinq-go/blob/master/src/rinq/revision.go#L141) only includes `Ref`